notebooks/demo/mg_jaccard.ipynb

```diff
@@ -0,0 +1,384 @@
+    "    cudf.set_option(\"spill\", True)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "enable_spilling()\n",
+    "cluster = LocalCUDACluster()\n",
+    "client = Client(cluster)\n",
+    "client.run(enable_spilling)\n",
+    "Comms.initialize(p2p=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Data\n",
+    "This notebook will use RMAT to generate a synthetic dataset.  The size of the dataset will be determined by the number of GPUs present.\n",
+    "This appraoch removes the need to find test data of various sizes.\n",
+    "\n",
+    "The notebook will call the cugraph.rmat function and have a dask_cudf DataFrame returned.  The rmat function could simply return a Graph object, which is more memory effecient, but the goal is to show the process starting with a dataframe\n",
+    "\n",
+    "| Number of GPUs | Scale | Edge Factor | Est Number of Nodes | Est Number of Edges |\n",
+    "|----------------|-------|-------------|---------------------|---------------------|\n",
+    "| 1              |  24   |     16      |      16,000,000     |      256,000,000    |\n",
+    "| 2              |  25   |     16      |      32,000,000     |      512,000,000    |\n",
+    "| 3              |  25   |     24      |      32,000,000     |      768,000,000    |\n",
+    "| 4              |  26   |     16      |      64,000,000     |    1,024,000,000    |\n",
+    "| 5              |  26   |     20      |      64,000,000     |    1,280,000,000    |\n",
+    "| 6              |  26   |     24      |      64,000,000     |    1,536,000,000    |\n",
+    "| 7              |  26   |     28      |      64,000,000     |    1,792,000,000    |\n",
+    "| 8              |  27   |     16      |     128,000,000     |    2,048,000,000    |"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "rmat_settings = (\n",
+    "    [24 , 16],\n",
+    "    [25 , 16],\n",
+    "    [25 , 24],\n",
+    "    [26 , 16],\n",
+    "    [26 , 20],\n",
+    "    [26 , 24],\n",
+    "    [26 , 28],\n",
+    "    [27 , 16],\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import subprocess\n",
+    "def get_gpu_memory_size():\n",
+    "    result = subprocess.check_output(\n",
+    "        [\n",
+    "            'nvidia-smi', '--query-gpu=memory.total'\n",
+    "            , '--format=csv,nounits,noheader'\n",
+    "        ]\n",
+    "    )    \n",
+    "    return result.decode('utf-8').strip().split('\\n')#).strip().split('\\n')\n",
+    "\n",
+    "gpu_info = get_gpu_memory_size()\n",
+    "number_of_gpus = len(gpu_info)\n",
+    "gpu_memory = int(gpu_info[0])\n",
+    "print(f\"the cluster has {number_of_gpus} GPUs where each GPU has {gpu_memory} GB of memory\")"
```

Comment:
sorry I don't have a recommendation for how to replace this yet, but I'd be surprised if there's not a dask client API that can give you this data (and more) without having to run a subprocess like this.  I don't know if this is recommended, bu perhaps something like this?
```
 |  memory_limit : int, float, str, or None, default "auto"
 |      Bytes of memory per process that the worker can use. Can be an integer (bytes),
 |      float (fraction of total system memory), string (like ``"5GB"`` or ``"5000M"``),
 |      or ``"auto"``, 0, or ``None`` for no memory management.
 |  device_memory_limit : int, float, str, or None, default 0.8
 |      Size of the CUDA device LRU cache, which is used to determine when the worker
 |      starts spilling to host memory. Can be an integer (bytes), float (fraction of
 |      total device memory), string (like ``"5GB"`` or ``"5000M"``), or ``"auto"``, 0,
 |      or ``None`` to disable spilling to host (i.e. allow full device memory usage).
...
>>> client.cluster.memory_limit
24579997696
>>> client.cluster.device_memory_limit
41231686041
```

Reply:
based on an offline discussion, this sounds like this is something we'd like to do but not in this PR.